interfaces/builtin/gsettings.go

```diff
@@ -0,0 +1,67 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/ubuntu-core/snappy/interfaces"
+)
+
+const gsettingsConnectedPlugAppArmor = `
+# Description: Can access global gsettings of the user's session. Restricted
+# because this gives privileged access to sensitive information stored in
+# gsettings and allows adjusting settings of other applications.
+# Usage: reserved
+
+#include <abstractions/dbus-session-strict>
+
+#include <abstractions/dconf>
+owner /{,var/}run/user/*/dconf/user w,
```

Comment:
Yes, that is absolutely a problem and not one which this PR is trying to address. It is important to remember this is for the 'global session gsettings' and as such, snaps need to use it as it is currently implemented, which is accessing /run/user/<uid>/dconf/user and /home/zyga/.config/dconf/user.

The path /home/zyga/snaps/$SNAP_NAME/$SNAP_REVISION/.config/dconf/user is unknown to gsettings as implemented and it cannot be used. Also, the future plan of apparmor gsettings mediation means that apps can continue to use the well-known paths (ie, not ~/snaps/...) and dconf will mediate access to snap-specific directories in the dconf database itself rather than using the actual filesystem.